### PR TITLE
Change hashtable to use OrdinalIgnoreCase to be case-insensitive in all Cultures

### LINKED ISSUE #7761: Powershell Hashtable Keys are case-sensitive on Ubuntu 18.04
On a clean installation of Ubuntu 18.04 from Azure, creating a hashtable in script results in a hashtable with case-sensitive keys.

The LANG environment variable is set to 'C.UTF-8' and both CultureInfo.CurrentCulture and CultureInfo.CurrentUICulture report  en-US-POSIX.

The result is scripts that are not case-strict when using hashtable keys will break.

Steps to reproduce
------------------
```powershell
$h = @{HostName = 'host'}
$h.hostname
$h.HostName
```

Expected behavior
-----------------
The value 'host' is returned for both $h.HostName and $h.hostname

Actual behavior
---------------
No value is returned for $h.HostName is 'host' while $h.hostname returns $null.

Environment data
----------------

```powershell
> $PSVersionTable
 Name                           Value
----                           -----
PSVersion                      6.1.0-rc.1
PSEdition                      Core
GitCommitId                    6.1.0-rc.1
OS                             Linux 4.15.0-1022-azure #23-Ubuntu SMP Thu Aug 16 15:19:42 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

Ubuntu18.04 seems to default to `C.UTF-8` for LANG (representing InvariantCulture) which results in a case-sensitive hashtable since CurrentCultureIgnoreCase doesn't work for that culture.  Fix is to use OrdinalIgnoreCase instead.

Fix https://github.com/PowerShell/PowerShell/issues/7761

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by lzybkr** [comment]
I'm pretty sure this is a breaking change.

**Comment by iSazonov** [comment]
> Ubuntu18.04 seems to default to C.UTF-8 for LANG (representing InvariantCulture)

Could you please reference where it come from and why?

Is there other cultures like `C.UTF-8` on Unix and Windows with the problem?

**Comment by SteveL-MSFT** [comment]
@iSazonov read the details in the related issue

**Comment by ffeldhaus** [comment]
@iSazonov In my opinion this pull request correctly fixes the issue described in #7761 and we should keep wider discussion of changing other culture-insensitive in other places out of this pull request.

As this is a serious issue in a core functionality (Hashtable) and as it is present on the default culture of Ubuntu 18.04 (C.utf8) and thus exposed to a large number of users in my opinion this should be merged with high priority and backport to 6.1.X and 6.0.X.

**Comment by iSazonov** [comment]
My request is not to block the PR. My fear is that there may be other places prone to this problem. This analysis may be useful for fixing other issues.

File: test/powershell/Language/Parser/LanguageAndParser.TestFollowup.Tests.ps1 [hunk 1]
```diff
@@ -303,3 +303,9 @@ Describe "Hash expression with if statement as value" -Tags "CI" {
         $hash['h'] | Should -BeExactly 'h'
     }
 }
+
+Describe "Hashtable is case insensitive" -Tag CI {
+    It "When LANG is C.UTF-8" -Skip:(!$IsLinux) {
```
Comment (ffeldhaus) [inline test/powershell/Language/Parser/LanguageAndParser.TestFollowup.Tests.ps1]: The issue affects Mac OS X as well, so please change to
```suggestion
    It "When LANG is C.UTF-8" -Skip:($IsWindows) {
```

**Review by ffeldhaus — CHANGES_REQUESTED** [review]
Issues also affects Mac OS, so test should run on Mac OS as well.

File: test/powershell/Language/Parser/LanguageAndParser.TestFollowup.Tests.ps1 [hunk 2]
```diff
@@ -303,3 +303,9 @@ Describe "Hash expression with if statement as value" -Tags "CI" {
         $hash['h'] | Should -BeExactly 'h'
     }
 }
+
+Describe "Hashtable is case insensitive" -Tag CI {
+    It "When LANG is C.UTF-8" -Skip:($IsWindows) {
+        sh -c 'LANG=C.UTF-8 pwsh -c ''$h=@{h=1};$h.H''' | Should -Be 1
```
Comment (bergmeister) [inline test/powershell/Language/Parser/LanguageAndParser.TestFollowup.Tests.ps1]: Minor nit: using the `-command` parameter on `pwsh` would be slightly more readable and using a different letter for the hashtable property.
```suggestion
        sh -c 'LANG=C.UTF-8 pwsh -command ''$h=@{p=1};$h.P''' | Should -Be 1
```

**Review by bergmeister — APPROVED** [review]
Looks good to me, only a minor, optional suggestion to improve readability of the test :)

**Review by ffeldhaus — APPROVED** [review]
Addresses the issue and test is good as well now!

**Review by anmenaga — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee reviewed this and agrees to take this change as the intention was to always have Hashtable keys case-insensitive

**Comment by TravisEz13** [comment]
@iSazonov 	Only one `CL-*` tag.  If it is a breaking change, don't put it in one of the others.

**Comment by adamedx** [comment]
Wow, thank you for fixing this. I am new to PowerShell Core and using it on Linux, and it seemed that this was intentional. I sent a lot of time fixing the casing of my hash tables. I'm guessing this is also the issue I ran into with sort-object, which gives different results on Linux and Windows:
```powershell
PS> $PSVersionTable.Platform;('a', 'ab', 'aC' | Sort-Object ) -join ','
Win32NT
a,ab,aC
PS> $PSVersionTable.Platform;('a', 'ab', 'aC' | Sort-Object ) -join ','
Unix
a,aC,ab
```

**Comment by ffeldhaus** [comment]
@adamedx I think it would be a good idea to track this in a separate issue and link it to #8064 where similar issues are being tracked.

**Pull request merged**